Add abandoned product ID to TransactionAbandon events

ianrumac opened the pull request:
## Changes in this pull request

- Adds `abandoned_product_id` property to customParameters in case of `TransactionAbandon` events

### Checklist

- [x] All unit tests pass.
- [x] All UI tests pass.
- [x] Demo project builds and runs.
- [x] I added/updated tests or detailed why my change isn't tested.
- [x] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [x] I have run `ktlint` in the main directory and fixed any issues.
- [x] I have updated the SDK documentation as well as the online docs.
- [x] I have reviewed the [contributing guide](https://github.com/superwall/Superwall-Android/tree/master/.github/CONTRIBUTING.md)

yusuftor commented:
Can you add a changelog entry pls. I added this to iOS

> SW-2901: Adds `abandoned_product_id` to a `transaction_abandon` event to use in audience filters. You can use this to show a paywall if a user abandons the transaction for a specific product.

But you'll need to change the linear ticket ID

ianrumac commented:
> Can you add a changelog entry pls. I added this to iOS
> 
> > SW-2901: Adds `abandoned_product_id` to a `transaction_abandon` event to use in audience filters. You can use this to show a paywall if a user abandons the transaction for a specific product.
> 
> But you'll need to change the linear ticket ID

Added, rebased first to refresh config to not cause any conflicts 👍
